### ISSUE TITLE
New match/case syntax reports "using variable before assignment"

### ISSUE THREAD

**Issue by devholland** [PyCQA/pylint#7388]
### Bug description

In the following code example, pylint reports: "*Using variable 'names' before assignment*":

```python
data = {'department': {'names': ['Steve', 'Anne']}}

match data['department'].get('names'):
    case names if names:
        for name in names:
            print(name)
    case _:
        print('no names found')
```

### Command used

```shell
Linted through VSCode integration.
```

### Pylint output

```shell
Using variable 'names' before assignment
```

### Expected behavior

The variable is assigned by the match/case statement and therefore shouldn't be reported as an error.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I'm going to close as it looks like a duplicate of https://github.com/PyCQA/pylint/issues/5327.
